File: enode/src/main/java/org/enodeframework/eventing/IEventSerializer.java
package org.enodeframework.eventing;

import java.util.List;
import java.util.Map;

/**
 * Represents an event.
 */
public interface IEventSerializer {
    /**
     * Serialize the given events to map.
     */
    Map<String, String> serialize(List<IDomainEvent<?>> evnts);

    /**
     * Deserialize the given data to events.
     */
    List<IDomainEvent<?>> deserialize(Map<String, String> data);
}

File: enode/src/main/java/org/enodeframework/eventing/IEventSerializer.kt
package org.enodeframework.eventing

/**
 * Represents an event.
 */
interface IEventSerializer {
    /**
     * Serialize the given events to map.
     */
    fun serialize(evnts: List<IDomainEvent<*>>): Map<String, String>

    /**
     * Deserialize the given data to events.
     */
    fun deserialize(data: Map<String, String>): List<IDomainEvent<*>>
}